labelme/app.py

```diff
@@ -875,12 +885,18 @@ def setDirty(self):
     def setClean(self):
         self.dirty = False
         self.actions.save.setEnabled(False)
-        self.actions.createMode.setEnabled(True)
-        self.actions.createRectangleMode.setEnabled(True)
-        self.actions.createCircleMode.setEnabled(True)
-        self.actions.createLineMode.setEnabled(True)
-        self.actions.createPointMode.setEnabled(True)
-        self.actions.createLineStripMode.setEnabled(True)
+        if self._classType == "object-2d" or self._classType is None:
+            self.actions.createRectangleMode.setEnabled(True)
+        else:
+            self.actions.createRectangleMode.setEnabled(False)
+        if self._classType == "segmentation" or self._classType is None:
+            self.actions.createMode.setEnabled(True)
+        else:
+            self.actions.createMode.setEnabled(False)
```

Comment:
```suggestion
        self.actions.createRectangleMode.setEnabled(False)
        self.actions.createMode.setEnabled(False)
        if self._classType == "object-2d":
            self.actions.createRectangleMode.setEnabled(True)
        if self._classType == "segmentation":
            self.actions.createMode.setEnabled(True)
```

이런식으로 일단 `False`로 초기화 해놓고 조건에 따라 `True`로 바꾸는건 어떨까요?

Reply:
조건에 들어갈 때 해당 함수가 두 번 호출되는건데 그것 보다는 else를 사용함으로서 한번씩만 호출되는게 더 좋을 것 같습니다.